compiler/executable/match_/planner/mod.rs

```diff
@@ -307,16 +307,23 @@ impl MatchExecutableBuilder {
     }
 
     fn push_check(&mut self, variables: &[Variable], check: CheckInstruction<ExecutorVariable>) {
+        // if it is a comparison or IID (TODO) we can inline the check into previous instructions
+        self.inline_as_optimisation(variables, &check);
```

Comment:
If this succeeds, we should probably short-circuit out of here.

Reply:
I think you're right - had this before and split it out again and I don't remember why... will do that